#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.14009441661142488 -0.9454226469223282 -0.2941930881592619 4.09415641082722
  position 0.8088228707250681 0.8448701608485496 -0.49839666784958037
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
DEF OMNI_WHEELS Robot {
  translation 0.2120658759860563 0.058577237796244845 -0.003606205714534046
  rotation 0.0028099568881768443 0.9999960517601033 -2.4627047272471087e-05 -2.031977718791124
  children [
    DEF SOLID1 Solid {
      children [
        DEF WHEEL1 HingeJoint {
          jointParameters HingeJointParameters {
            position 16.417520778494396
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            RotationalMotor {
              name "motor_1"
              maxVelocity 30.3688
              maxTorque 0.847386
            }
            PositionSensor {
              name "position_1"
              resolution 0.00384841
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 1 3.851150164135223
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  position -949.7730152325737
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation 1 0 0 5.2711514587234465
                  children [
                    DEF r1 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        DEF SMALL_WHEEL_SHAPE Shape {
                          appearance PBRAppearance {
                            baseColor 0.8 0 0
                            roughness 1
                            metalness 0
                          }
                          geometry Cylinder {
                            height 0.02
                            radius 0.01
                          }
                        }
                      ]
                    }
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  position -3644.2492074691313
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation -0.7071067811865476 -0.7071067811865476 0 0.001729304971045463
                  children [
                    DEF r2 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  position 2999.198883673491
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation -0.9999999999999999 0 0 4.163693158351482
                  children [
                    DEF r3 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  position 2988.627268231559
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation -0.7071067811865476 -0.7071067811865476 0 2.1689379859242623
                  children [
                    DEF r4 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  position -2873.353429835863
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  rotation 0 1 0 4.345440852387453
                  children [
                    DEF r5 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  position 2970.756389661638
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  rotation 0 0.9999999999999999 0 5.092924672873151
                  children [
                    DEF r6 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  position -6713.770383992831
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation 0.7071067811865476 -0.7071067811865476 0 3.3284759250324725
                  children [
                    DEF r7 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  position 1473.3500466527255
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation -0.7071067811865475 0.7071067811865475 0 3.08468477270226
                  children [
                    DEF r8 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF WHEEL_TRANS Transform {
                translation 0 0 0.02
                rotation 1 0 0 1.5708
                children [
                  DEF WHEEL_SHAPE Shape {
                    appearance PBRAppearance {
                      baseColor 0.1803921568627451 0.20392156862745098 0.21176470588235294
                      roughness 1
                      metalness 0
                    }
                    geometry Cylinder {
                      height 0.02
                      radius 0.05
                      subdivision 20
                    }
                  }
                ]
              }
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "wheel 1"
      boundingObject DEF DUMMY Sphere {
        radius 0.01
      }
      physics Physics {
      }
    }
    DEF SOLID2 Solid {
      rotation 0 1 0 2.09
      children [
        DEF WHEEL2 HingeJoint {
          jointParameters HingeJointParameters {
            position 33.74351158629518
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            RotationalMotor {
              name "motor_2"
              maxVelocity 30.3688
              maxTorque 0.847386
            }
            PositionSensor {
              name "position_2"
              resolution 0.00384841
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 -0.9999999999999999 3.9531944467825664
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  position 329.3188913729423
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation 1 0 0 2.5932553996037924
                  children [
                    USE r1
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  position -524.5076781619869
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation 0.7071067811865475 0.7071067811865475 0 3.279887641098404
                  children [
                    USE r2
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  position -1567.6305286197337
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation 1 0 0 3.165798175162924
                  children [
                    USE r3
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  position -1799.814476369829
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation -0.7071067811865475 -0.7071067811865475 0 2.8234785164672482
                  children [
                    USE r4
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  position -295.1437920635754
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  rotation 0 -0.9999999999999999 0 6.117267933314425
                  children [
                    USE r5
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  position -1506.3922642689502
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  rotation 0 0.9999999999999999 0 1.572209454150597
                  children [
                    USE r6
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  position 2136.7217442626875
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0.4387398216280778
                  children [
                    USE r7
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  position -5216.49243117157
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation 0.7071067811865476 -0.7071067811865476 0 1.448626212513612
                  children [
                    USE r8
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              USE WHEEL_TRANS
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "wheel 2"
      boundingObject USE DUMMY
      physics Physics {
      }
    }
    DEF SOLID3 Solid {
      rotation 0 1 0 4.18
      children [
        DEF WHEEL3 HingeJoint {
          jointParameters HingeJointParameters {
            position -7.545909814211655
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            RotationalMotor {
              name "motor_3"
              maxVelocity 30.3688
              maxTorque 0.847386
            }
            PositionSensor {
              name "position_3"
              resolution 0.00384841
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 1 5.0243702401471975
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  position -931.4856338329479
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation -1 0 0 1.5742083703690846
                  children [
                    USE r1
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  position -1949.8941748804111
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation -0.7071067811865475 -0.7071067811865475 0 2.1067296547393206
                  children [
                    USE r2
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  position 179.76698042627584
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation 0.9999999999999999 0 0 3.837791825247415
                  children [
                    USE r3
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  position -10.620909281391585
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation 0.7071067811865475 0.7071067811865475 0 1.945461332967588
                  children [
                    USE r4
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  position -2273.1443005334377
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  rotation 0 1 0 1.3687806655726324
                  children [
                    USE r5
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  position -1983.3695733204158
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  rotation 0 1 0 2.116983748333488
                  children [
                    USE r6
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  position -3444.4449807828
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation 0.7071067811865476 -0.7071067811865476 0 1.259432448386641
                  children [
                    USE r7
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  position -426.17074594786203
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 1.0858549403498496
                  children [
                    USE r8
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              USE WHEEL_TRANS
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "wheel 3"
      boundingObject USE DUMMY
      physics Physics {
      }
    }
    DEF BODY Transform {
      translation 0 0.03 0
      rotation 0 0 1 4.712390306100425
      scale 1 0.7 0.7
      children [
        Shape {
          appearance Appearance {
            material Material {
              ambientIntensity 1
              diffuseColor 0.7568627450980392 0.49019607843137253 0.06666666666666667
              shininess 0.4
              specularColor 0.396078 0.409804 0.466667
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                0 -0.159225 0.085098
                0 -0.153309 -0.095344
                0 -0.143127 0.076928
                0 -0.138185 -0.085488
                0 -0.005915 0.180442
                0 -0.004942 0.162416
                0 0.004942 -0.162416
                0 0.005915 -0.180442
                0 0.138185 0.085488
                0 0.143127 -0.076928
                0 0.153309 0.095344
                0 0.159225 -0.085098
                0.03 -0.143127 0.076928
                0.03 -0.138185 -0.085488
                0.03 -0.004942 0.162416
                0.03 0.004942 -0.162416
                0.03 0.138185 0.085488
                0.03 0.143127 -0.076928
                0.05 -0.159225 0.085098
                0.05 -0.153309 -0.095344
                0.05 -0.005915 0.180442
                0.05 0.005915 -0.180442
                0.05 0.153309 0.095344
                0.05 0.159225 -0.085098
              ]
            }
            coordIndex [
              20, 22, 4, -1, 4, 22, 10, -1, 18, 20, 0, -1, 0, 20, 4, -1, 19, 18, 1, -1, 1, 18, 0, -1, 21, 19, 7, -1, 7, 19, 1, -1, 23, 21, 11, -1, 11, 21, 7, -1, 22, 23, 10, -1, 10, 23, 11, -1, 23, 22, 21, -1, 21, 22, 20, -1, 21, 20, 19, -1, 19, 20, 18, -1, 6, 7, 3, -1, 3, 7, 1, -1, 3, 1, 2, -1, 2, 1, 0, -1, 2, 0, 5, -1, 5, 0, 4, -1, 5, 4, 8, -1, 8, 4, 10, -1, 8, 10, 9, -1, 9, 10, 11, -1, 9, 11, 6, -1, 6, 11, 7, -1, 17, 16, 9, -1, 9, 16, 8, -1, 16, 14, 8, -1, 8, 14, 5, -1, 14, 12, 5, -1, 5, 12, 2, -1, 12, 13, 2, -1, 2, 13, 3, -1, 13, 15, 3, -1, 3, 15, 6, -1, 15, 17, 6, -1, 6, 17, 9, -1, 15, 13, 17, -1, 17, 13, 12, -1, 17, 12, 16, -1, 16, 12, 14, -1
            ]
          }
        }
      ]
    }
    Solid {
      translation 0.1 0.036 0.060000000000000005
      rotation 0 1 0 -0.6544996938995746
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.9372549019607843 0.1607843137254902 0.1607843137254902
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.1
          }
        }
      ]
    }
    DistanceSensor {
      translation 0.13 0.04 0.019999999999999997
      rotation 0 1 0 -0.43633269389957474
      name "distance_sensor2"
      lookupTable [
        0 0 0
        0.2 65535 0
      ]
      resolution 1
    }
    DistanceSensor {
      translation 0.06999999999999999 0.04 0.09999999999999999
      rotation 0 1 0 -0.5672326938995748
      name "distance_sensor1"
      lookupTable [
        0 0 0
        0.2 65535 0
      ]
      aperture 1.57
      resolution 1
    }
  ]
  boundingObject USE BODY
  physics Physics {
  }
  controller "U2_Exam_Victor_Xix"
}
SoccerBall {
  translation -0.5780825419164315 0.11205823999999993 0.4445004102638695
  rotation -0.5388163019190297 0.46552249174604765 -0.7021152344628709 -1.5069488002103555
}
